int32     state
float32[] pose
float32[] trans